src/containers/App.jsx

```diff
@@ -93,48 +86,28 @@ const App = ({ accounts, konnectors, triggers }) => {
   const shortcutsDirectories = canHaveShortcuts
     ? formatShortcuts(folders, customHomeShortcuts)
     : null
-  const context = useQuery(contextQuery.definition, contextQuery.options)
+
+  const { isFetching, hasError } = useFetchInitialData()
 
   const showAssistantForMobile = isFlagshipApp()
     ? flag('cozy.searchbar.enabled-for-flagship')
     : flag('cozy.searchbar.enabled') && isMobile
 
-  useEffect(() => {
-    setIsFetching(
-      [accounts, konnectors, triggers, context].some(collection =>
-        ['pending', 'loading'].includes(collection.fetchStatus)
-      )
-    )
-    setHasError(
-      [accounts, konnectors, triggers, context].find(
-        collection => collection.fetchStatus === 'failed'
-      )
-    )
-  }, [accounts, konnectors, triggers, context])
-
-  if (context?.attributes?.features) {
-    const flags = toFlagNames(context.attributes.features)
-    enableFlags(flags)
-  }
-
-  useEffect(() => {
-    setIsReady(
-      appsReady &&
-        !hasError &&
-        !isFetching &&
-        shortcutsDirectories !== undefined
-    )
-  }, [appsReady, hasError, isFetching, shortcutsDirectories])
-
-  useEffect(() => {
-    if (isReady && webviewIntent) {
+  if (
+    !didInit &&
+    !hasError &&
+    !isFetching &&
+    shortcutsDirectories !== undefined
+  ) {
+    if (webviewIntent) {
       webviewIntent.call('setTheme', theme.variant)
       webviewIntent.call('hideSplashScreen')
     }
-    if (isReady && !webviewIntent && __SIMULATE_FLAGSHIP__) {
+    if (!webviewIntent && __SIMULATE_FLAGSHIP__) {
       document.getElementById('splashscreen').style.display = 'none'
     }
-  }, [isReady, theme, webviewIntent])
+    setDidInit(true)
+  }
```

Comment:
Can't we use a solution without any state ? Like 
`const isReady = !hasError && !isFetching && shortcutsDirectories !== undefined`

Reply:
Sounds smart :+1: